app/controllers/taxon_changes_controller.rb

```diff
@@ -404,7 +417,27 @@ def analyze_ids
       format.json { render json: response[:json], status: response[:status] }
     end
   end
+
+  def withdraw
+    @taxon_change = TaxonChange.find( params[:id] )
+    if @taxon_change.status_withdrawn!
+      flash[:notice] = "Taxon change was withdrawn."
+    else
+      flash[:error] = "Could not withdraw taxon change."
+    end
+    redirect_to edit_taxon_change_path( @taxon_change )
+  end
   
+  def restore
+    @taxon_change = TaxonChange.find(params[:id])
+    if @taxon_change.update(status: :draft)
+      flash[:notice] = "Taxon change was restored to draft."
```

Comment:
I think we should add a model validation to TaxonChange so that if committed_on is not null, it cannot have a status other than `committed`. Otherwise this endpoint could be called to make an already committed taxon change a draft, and then later the UI is only checking the status column and not the committed_on column, so the UI offers the ability to commit it again. Or the above endpoint can mark a committed change as withdrawn. Since only `status` is being checked for some things and `committed_on` not checked, we should have model validations to ensure the two attributes stay in sync

Reply:
I added some validations and tests